test/integration/apiserver/apply/data.go

```diff
@@ -0,0 +1,676 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserver
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/kubernetes/test/integration/etcd"
+)
+
+var resetFieldsStatusData = map[schema.GroupVersionResource]string{
+	gvr("", "v1", "services"): `{"status": {"loadBalancer": {"ingress": [{"ip": "127.0.0.2"}]}}}`,
+}
+
+const resetFieldsStatusDefault = `{"status": {"conditions": [{"type": "MyStatus", "status":"False"}]}}`
+
+func getResetFieldsEtcdStorageData() map[schema.GroupVersionResource]etcd.StorageData {
+	return getResetFieldsEtcdStorageDataForNamespace("etcdstoragepathtestnamespace")
+}
+
+// getResetFieldsTestEtcdStorageDataForNamespace is a copy of etcd.GetEtcdStorageDataForNamespace with
+// exactly one field in the spec and one field in the status different in order to test resetFields.
+func getResetFieldsEtcdStorageDataForNamespace(namespace string) map[schema.GroupVersionResource]etcd.StorageData {
+	//image := image.GetE2EImage(image.BusyBox)
+	etcdStorageData := map[schema.GroupVersionResource]etcd.StorageData{
+		// k8s.io/kubernetes/pkg/api/v1
+		gvr("", "v1", "services"): {
+			Stub:             `{"metadata": {"name": "service1"}, "spec": {"externalName": "service2name", "ports": [{"port": 10000, "targetPort": 11000}], "selector": {"test": "data"}}}`,
```

Comment:
shouldn't there be two objects here?

Reply:
the other object is in `test/integration/etcd/data.go` unless I'm misunderstanding.

Reply:
I see, that is going to make this test really hard to read though?

Reply:
er, hm. do we need a separate pair for each cleared field? that would be unfortunate.

Reply:
No we shouldn't need a separate pair per field, just two different objects that have at least one field in the spec and status different from that field on the other object.

I agree that it would be more readable if both pairs were in the same file locally in the apply package. That would also probably alleviate the potential issue where someone changes `test/integration/etcd/data.go` and doesn't understand why the broke the unrelated reset fields test.

I guess the tradeoff is that it's probably not great to have two copies of all this junk and that they could potentially diverge? I'm fine either way if you have a preference?

Reply:
OK I had an idea to make this readable: make the 2nd object contain *just* the field to be changed, and then in code to make the second object, first unmarshal the first object's json, and then unmarshal the second object's json *on top of it*. WDYT?

Reply:
it's still not clear to me why you think we only need one of the reset fields rather than all of them somehow? Don't we need to verify that none of them accidentally get owned?

Reply:
(I agree any field under status is sufficient to test status. but some of the types had various separate fields under spec IIRC)